add assertion to check existence of shareholder in the contract

sjoshisupra opened the pull request:
Add assertion to check that a shareholder exist in the contract as a precondition to `vest_individual` to avoid failure in `simple_map::borrow`.

sjoshisupra committed:
add assertion to check existence of shareholder in the contract

nizam-supraoracles reviewed (commented):
I suggest we add the `assert_shareholder_exists` condition inside the `remove_shareholder` function too.

sjoshisupra committed:
add assertion in remove_shareholder

sjoshisupra commented:
> I suggest we add the `assert_shareholder_exists` condition inside the `remove_shareholder` function too.

addressed in the new commit

axiongsupra commented on aptos-move/framework/supra-framework/sources/vesting_without_staking.move:
```diff
@@ -383,17 +383,19 @@ module supra_framework::vesting_without_staking {
     }
 
 	public entry fun vest_individual(contract_address: address, shareholder_address: address) acquires VestingContract {
-		assert_active_vesting_contract(contract_address);
```
@sjoshisupra why we removed this check as the vesting contract can be terminated at the end of `vest_individual`. It will cause a problem when call vest method who will call `vest_individual` in a loop.

axiongsupra reviewed (commented):
Thanks for add the check

sjoshisupra committed:
move termination logic

move termination logic to `vest` instead of `vest_individual`. This is done
to prevent following from happening:
If 5 shareholders are there and only 2 have non-zero balance when `vest` is called
after these 2 get their vesting due, if `vest_individual` sets the termination flag,
in the loop when `vest_individual` is called for 3rd shareholder assertion will fail
and the whole Tx will abort.
